cmod/conditions.py

```diff
@@ -0,0 +1,226 @@
+        cmd.board.add_lumi_vis_separation(id, 5, h)
+
+        self.gantry_conditions["lumi_vs_FOV_center"]["diff"] = (self.gantry_conditions["lumi_vs_FOV_center"]["diff"]*self.gantry_conditions["lumi_vs_FOV_center"]["num_sipms"] + h) /  self.gantry_conditions["lumi_vs_FOV_center"]["num_sipms"]+1
+
+        self.gantry_conditions["lumi_vs_FOV_center"]["num_sipms"] += 1
+
+    else:
+      for id in range(1, len(cmd.board.detectors)+1):
+        self.calculate_sipm_vis_coordinates(cmd, id, False)
+    
+    if save_gantry_conditions_changes:
+      self.save_gantry_conditions()
+
+
+  def calculate_sipm_lumi_coordinates(self, cmd, detid=None, save_gantry_conditions_changes=True):
+    if detid is not None:
+      halign_cmd_str = """halign --detid={detid}
+            --channel={channel} --mode={mode}
+            --sample={samples} -z {zval}  --overwrite
+            --range={range} --distance={distance}
+            --power={power}
+            --wipefile
+        """.format(detid=id,
+            mode= cmd.board.get_detector(id).mode,
+            channel= cmd.board.get_detector(id).channel,
+            samples= 100,
+            zval= 10,
+            range= 6,
+            distance= 2,
+            power= 0.5)
+        
+      command2 = cmd.precmd(halign_cmd_str)
+      sig2 = cmd.onecmd(command2)
+      status2 = cmd.postcmd(sig2, command2)
+
+      if cmd.board.vis_coord_hasz(id, 5):
+        h = cmd.board.get_lumi_coord(id, 5)-cmd.board.get_vis_coord(id, 5)
+        cmd.board.add_lumi_vis_separation(id, 5, h)
+
+        self.gantry_conditions["lumi_vs_FOV_center"]["diff"] = (self.gantry_conditions["lumi_vs_FOV_center"]["diff"]*self.gantry_conditions["lumi_vs_FOV_center"]["num_sipms"] + h) /  self.gantry_conditions["lumi_vs_FOV_center"]["num_sipms"]+1
+
+        self.gantry_conditions["lumi_vs_FOV_center"]["num_sipms"] += 1
+
+    else:
+      for id in range(1, len(cmd.board.detectors)+1):
+        self.calculate_sipm_lumi_coordinates(cmd, id, False)
+    
+
+    if save_gantry_conditions_changes:
+      self.save_gantry_conditions()
+
+  # increments the use count
+  def increment_use_count(self):
+    self.gantry_conditions.gc_use_count += 1
+    # update the use count antry_ionditionsn the latest conditions file
+    # get the latest conditions file
+    filename = Conditions.get_latest_gantry_conditions_filename()
+    # save the conditions to the file
+    self.save_gantry_conditions(filename)
+
+  # define get, calculate functions for the data quality(long term) conditions and the board conditions
+  def get_board_conditions(self):
+    pass
+  
+  def calculate_board_conditions(self):
+    pass
+
+  # TODO: a function to load data quality(long term) conditions from a file
+  # TODO: a function to save data quality(long term) conditions to a file
+  # TODO: a getter for the data quality(long term) conditions
+  # TODO: implement the data quality(long term) conditions calculation
+  def get_data_quality_conditions(self):
```

Comment:
We probably want to make a new command for this, to the CLI users can call it whenever. 

You can put it in the `getset.py`

Reply:
Also, do you think it might be better for us to make 2 separate issues for implementing the gantry conditions and another one for implementing the data quality conditions? So that I can address those issues after we have enough info?

Reply:
Pardon the confusion. Data quality condition is only used for the post-processing of data. This does not need to be implemented in the tileboard.